packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-group.ts

```diff
@@ -0,0 +1,196 @@
+
+  constructor(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupProps) {
+    super(scope, id);
+
+    let serviceRole: iam.Role | undefined = props.serviceRole;
+    if (serviceRole) {
+      if (serviceRole.assumeRolePolicy) {
+        serviceRole.assumeRolePolicy.addStatement(new iam.PolicyStatement()
+          .addAction('sts:AssumeRole')
+          .addServicePrincipal('codedeploy.amazonaws.com'));
+      }
+    } else {
+      serviceRole = new iam.Role(this, 'ServiceRole', {
+        assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com')
+      });
+    }
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('arn:aws:s3:::*/CodeDeploy/*')
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('*')
+      .addCondition('StringEquals', {
+        's3:ExistingObjectTag/UseWithCodeDeploy': 'true'
+      })
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource(props.alias.functionArn)
+      .addActions('lambda:UpdateAlias', 'lambda:GetAlias'));
+    if (props.alarms) {
+      serviceRole.addToPolicy(new iam.PolicyStatement()
+        .addResources(...props.alarms.map(alarm => alarm.alarmArn))
+        .addAction('cloudwatch:DescribeAlarms'));
+    }
+
+    const alarms = props.alarms || [];
+
+    const resource = new CfnDeploymentGroup(this, 'Resource', {
+      applicationName: props.application.applicationName,
+      serviceRoleArn: serviceRole.roleArn,
+      deploymentGroupName: props.deploymentGroupName,
+      deploymentConfigName: `CodeDeployDefault.Lambda${props.deploymentConfig}`,
+      deploymentStyle: {
+        deploymentType: props.deploymentConfig === LambdaDeploymentConfig.AllAtOnce ?
+          DeploymentType.InPlace : DeploymentType.BlueGreen,
+        deploymentOption: props.deploymentConfig === LambdaDeploymentConfig.AllAtOnce ?
+          DeploymentOption.WithoutTrafficControl : DeploymentOption.WithTrafficControl
+      },
+
+      alarmConfiguration: new cdk.Token(() => renderAlarmConfiguration(alarms, props.ignorePollAlarmsFailure)),
+      autoRollbackConfiguration: new cdk.Token(() => renderAutoRollbackConfiguration(alarms, props.autoRollback)),
+    });
+
+    this.application = props.application;
+    this.deploymentGroupName = resource.deploymentGroupName;
+    this.deploymentGroupArn = deploymentGroupNameToArn(this.application.applicationName, this.deploymentGroupName, this);
+
+    if (props.preHook) {
+      this.grantPutLifecycleEventHookExecutionStatus(props.preHook.role);
+      props.preHook.grantInvoke(serviceRole);
+    }
+    if (props.postHook) {
+      this.grantPutLifecycleEventHookExecutionStatus(props.postHook.role);
+      props.postHook.grantInvoke(serviceRole);
+    }
+
+    (props.alias.node.findChild('Resource') as lambda.CfnAlias).options.updatePolicy = {
```

Comment:
We are going to add a `cfnresource` to L2s so you'd be able to access it without an escape hatch. Feel free to do that now or raise an issue not to forget this escape hatch later.